modules/management/src/main/java/org/mule/module/management/agent/AbstractJmxAgent.java

```diff
@@ -427,9 +427,9 @@ protected void registerServiceServices() throws NotCompliantMBeanException, MBea
     protected void registerFlowConstructServices() throws NotCompliantMBeanException, MBeanRegistrationException,
         InstanceAlreadyExistsException, MalformedObjectNameException
     {
-        for (AbstractFlowConstruct flowConstruct : muleContext.getRegistry().lookupObjects(AbstractFlowConstruct.class))
+        for (String  rawName : muleContext.getRegistry().lookupByType(AbstractFlowConstruct.class).keySet())
```

Comment:
Currently that is not a problem as only the managed beans for the flow type tries to retrieve something from the spring registry in its postconstruct. Anyway, I've pushed changes so that the case of connectors and model/services use the spring bean definition name as the name of the jmx managed bean. This is for coherence purposes. Connectors cannot set placeholders in its names as the name property in mule.xsd is of substitutableName type which has a restriction propertyPlaceHolderType.